server/src/main/java/com/saralein/server/router/Router.java

```diff
@@ -2,74 +2,40 @@
 
 import com.saralein.server.controller.Controller;
 import com.saralein.server.controller.ErrorController;
-import com.saralein.server.protocol.Methods;
+import com.saralein.server.middleware.Callable;
 import com.saralein.server.request.Request;
-import com.saralein.server.response.*;
-import java.nio.file.FileSystems;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
+import com.saralein.server.response.Response;
 
-public class Router {
+public class Router implements Callable {
     private final Routes routes;
-    private final Path root;
-    private final Controller directoryController;
-    private final Controller fileController;
     private final ErrorController errorController;
-    private boolean resourceExists = false;
-    private boolean resourceIsDirectory = false;
 
-    public Router(
-            Controller directoryController,
-            Controller fileController,
-            ErrorController errorController,
-            Routes routes, Path root) {
-        this.directoryController = directoryController;
-        this.fileController = fileController;
-        this.errorController = errorController;
-        this.routes = routes;
-        this.root = root;
-    }
-
-    public Response resolveRequest(Request request) {
-        resourceStatus(request);
-        Controller controller = resourceExists ? routeResource(request) : routeUri(request);
-
-        return controller.respond(request);
-    }
-
-    private void resourceStatus(Request request) {
-        String separator = FileSystems.getDefault().getSeparator();
-        String resourceUri = root.toAbsolutePath() + separator + request.getUri();
-        Path resource = Paths.get(resourceUri);
-        resourceExists = Files.exists(resource);
-        resourceIsDirectory = Files.isDirectory(resource);
+    public Router(Routes routes) {
+        this(routes, new ErrorController());
     }
 
-    private Controller routeResource(Request request) {
-        String requestMethod = request.getMethod();
-        String allowedMethods = Methods.allowedFileSystemMethods();
-        boolean isAllowedMethod = allowedMethods.contains(requestMethod);
-
-        if (resourceIsDirectory && isAllowedMethod) {
-            return directoryController;
-        } else if (isAllowedMethod) {
-            return fileController;
-        } else {
-            return errorController.updateStatus(405);
-        }
+    public Router(Routes routes, ErrorController errorController) {
+        this.routes = routes;
+        this.errorController = errorController;
```

Comment:
So it seems a little off to be using a controller outside of routing, maybe we can abstract error handling into another piece of middleware?

Reply:
Yes, I started addressing this on the next PR.